e2etest/webhookstests/webhooks.go

```diff
@@ -106,6 +113,51 @@ var _ = ginkgo.Describe("Webhooks", func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(ContainSubstring("overlaps with already defined CIDR"))
 		})
+
+		ginkgo.DescribeTable("IPAddressPool with overlapping addresses of the nodes",
+			func(ipFamily ipfamily.Family) {
+				nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				nodeIps, err := k8s.NodeIPsForFamily(nodes.Items, ipFamily, "")
+				Expect(err).NotTo(HaveOccurred())
+				for _, nodeIPString := range nodeIps {
+					nodeIP := net.ParseIP(nodeIPString)
+					var nodeCIDRs []string
+					if nodeIP.To4() != nil {
```

Comment:
we now the family already, no need to do this if. Also, it should be enough just to pick the first node, no need to iterate over all

Reply:
not sure I understand you. do you mean this if check at line 126? This check is needed to differentiate ipv4 and ipv6 since they have different CIDRMask. 

I agree, we don't need to iterate all nodes ip, first one is enough. will change.

Reply:
what I mean is, we have ipFamily as parameter, so instead of parsing the node ip and trying to understand if it's v4 or v6, we should rely on that.

Reply:
ok. what happen if ipfamily is ipfamily.DualStack? it could return both ipv4 and ipv6 nodeIPs.

Reply:
ipfamily is a parameter we are driving here, and we have only v4 and v6 iiuc

Reply:
I mean ipFamily is what you are setting here, https://github.com/metallb/metallb/pull/2585/files#diff-d32fb1c70f00b7179f34c7f6d38e974556efc26ffa22cc0d808991e07f13140eR158 in line 158